internal/kibana/maintenance_window/delete.go

```diff
@@ -0,0 +1,43 @@
+package maintenance_window
+
+import (
+	"context"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/kibana_oapi"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+)
+
+func (r *MaintenanceWindowResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var stateModel MaintenanceWindowModel
+
+	diags := req.State.Get(ctx, &stateModel)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	serverVersion, sdkDiags := r.client.ServerVersion(ctx)
+	if sdkDiags.HasError() {
+		return
+	}
+
+	serverFlavor, sdkDiags := r.client.ServerFlavor(ctx)
+	if sdkDiags.HasError() {
+		return
+	}
+
+	diags = validateMaintenanceWindowServer(serverVersion, serverFlavor)
```

Comment:
We've typically only checked the server version during creations (or updates if it relates to a specific attribute). It's possible that a user has changed the underlying stack, and downgraded to an unsupported version, but it's pretty unlikely and the delete would result in a 404 which would then correctly result in the resource being removed from state. 

I don't mind leaving this in, but I don't think it's required.

Reply:
Makes sense, removed.